conan/tools/gnu/makedeps.py

```diff
@@ -0,0 +1,602 @@
+        """
+        def _update_make_files(info):
+            make_files[f"{info.name}.mk"] = self._content_generator.content(info)
+            for alias in info.aliases:
+                alias_info = _MakeInfo(alias, [info.name], f"Alias {alias} for {info.name}", None, [])
+                make_files[f"{alias}.mk"] = self._content_generator.package_content(alias_info)
+
+        make_files = {}
+        # If the package has no components, then we have to calculate only the root makefile
+        if not self._dep.cpp_info.has_components:
+            package_info = self.package_info
+            _update_make_files(package_info)
+            return make_files
+
+        # First, let's load all the components makefile files
+        # Loop through all the package's components
+        pkg_requires = []
+        for component_info in self.components_info:
+            _update_make_files(component_info)
+            # Saving components name as the package requires
+            pkg_requires.append(component_info.name)
+
+        # Second, let's load the root package's makefile file ONLY
+        # if it does not already exist in components one
+        pkg_name = _get_package_name(self._dep, self._build_context_suffix)
+        if f"{pkg_name}.mk" not in make_files:
+            package_info = _MakeInfo(pkg_name, pkg_requires, f"Conan package: {pkg_name}", None, _get_package_aliases(self._dep))
+            # It'll be enough creating a makefile. This file will be like an alias
+            make_files[f"{package_info.name}.mk"] = self._content_generator.package_content(package_info)
+            for alias in package_info.aliases:
+                alias_info = _MakeInfo(alias, [package_info.name], f"Alias {alias} for {package_info.name}", None, [])
+                make_files[f"{alias}.mk"] = self._content_generator.package_content(alias_info)
+
+        return make_files
+
+
+class _MakeGlobalGenerator:
+    """
+    Generate and stare the conandeps.mk file
+    """
+
+    _output_filename = "conandeps.mk"
+
+    def __init__(self, conanfile, dep, reqs):
+        """
+        :param conanfile: The current recipe object
+        :param dep: The current dependency object
+        :param reqs: The current dependency requirements
+        """
+        self._content_global_generator = _MakeGlobalContentGenerator(conanfile, dep)
+        self._reqs = reqs
+
+    @property
+    def make_file(self):
+        """
+        Get the .mk file name and its content
+        """
+        return {self._output_filename: self._content_global_generator.content(self._reqs)}
+
+
+class MakeDeps(object):
+    """
+    Generate and store all the .mk files for all the dependencies
+    """
+
+    def __init__(self, conanfile):
+        """
+        :param conanfile: The current recipe object
+        """
+        self._conanfile = conanfile
+        self.build_context_activated = []
+        self.build_context_suffix = {}
```

Comment:
We can drop the build context at the moment, lets start small and simpler with host context first

Reply:
Okay, makes sense. Most of this generator I made based on pkg-config deps, that's a very complete generator, but I forgot a very important information, pkg-config is consumed by autotools, but make deps is something totally separated.